### PR TITLE
docs: Move content on rootless mode from operator manual

### PR DESCRIPTION
**Pull request by vikram-dagger**
This is branched off https://github.com/dagger/dagger/pull/9022 so should be merged after.

### CODE REVIEW

**Comment by jedevc** [comment]
@vikram-dagger I think this one can be rebased now onto main, since #9022 is merged.

**Comment by vikram-dagger** [comment]
Rebased

File: docs/current_docs/configuration/security.mdx
Comment (jedevc) [inline docs/current_docs/configuration/security.mdx:1]: This is an empty file now?

**Review by jedevc — APPROVED** [review]

File: docs/current_docs/configuration/security.mdx
Comment (vikram-dagger) [reply docs/current_docs/configuration/security.mdx:1]: Good catch, thanks!

**Pull request merged**